### ISSUE TITLE
Clarity Modal announcing complete modal from screen reader(NVDA) as soon as it opens

### ISSUE THREAD

**Issue by anusha582** [vmware-clarity/ng-clarity#1991]
Please follow below steps:

1) Open screen reader (NVDA) 
2) Click on any button that opens the Modal - refer the screenshot from clarity
3) Screen reader is announcing complete content and no focus on X mark 
4) Another issue  reading as Clickable even for non-interactive element what ever we add inside modal-body block.

Technology In my project
1. Angular 17 
2."@clr/angular": "^17.4.0",
    "@clr/icons": "^13.0.2",
    "@clr/ui": "^17.4.0"

<img width="1902" height="842" alt="Image" src="https://github.com/user-attachments/assets/0c022d80-fa87-4b3d-bbeb-e387ffc85b25" />
